pkg/workloadmeta/store.go

```diff
@@ -318,21 +318,35 @@ func (s *store) ListProcessesWithFilter(filter ProcessFilterFunc) []*Process {
 
 // GetKubernetesPodForContainer implements Store#GetKubernetesPodForContainer
 func (s *store) GetKubernetesPodForContainer(containerID string) (*KubernetesPod, error) {
-	entities, ok := s.store[KindKubernetesPod]
+	s.storeMut.RLock()
+	defer s.storeMut.RUnlock()
+
+	containerEntities, ok := s.store[KindContainer]
 	if !ok {
 		return nil, errors.NewNotFound(containerID)
 	}
 
-	for _, e := range entities {
-		pod := e.cached.(*KubernetesPod)
-		for _, podContainer := range pod.Containers {
-			if podContainer.ID == containerID {
-				return pod, nil
-			}
-		}
+	containerEntity, ok := containerEntities[containerID]
+	if !ok {
+		return nil, errors.NewNotFound(containerID)
+	}
+
+	container := containerEntity.cached.(*Container)
+	if container.Owner == nil || container.Owner.Kind != KindKubernetesPod {
+		return nil, errors.NewNotFound(containerID)
+	}
+
+	podEntities, ok := s.store[KindKubernetesPod]
+	if !ok {
+		return nil, errors.NewNotFound(container.Owner.ID)
```

Comment:
Nit: let's say there is a bug and the content of `container` isn't what's expected after the L329 read in the map, especially the `container.Owner.ID` field. This error then reports on an ID which isn't the same as the one passed to the method call.
I would thus prefer a:
```suggestion
		return nil, errors.NewNotFound(containerID)
```

That's purely debug/troubleshooting-wise, and just a nit. Discard if you feel like it.

Reply:
Indeed, it is not the same but IIRC the pod ID has a different format from the container ID. It makes it easier to identify whether the pod or the container is not found.